sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/DefaultSource.scala

```diff
@@ -61,7 +61,9 @@ class DefaultSource extends FileFormat with DataSourceRegister {
     val firstRow = new LineCsvReader(csvOptions).parseLine(firstLine)
 
     val header = if (csvOptions.headerFlag) {
-      firstRow
+      firstRow.zipWithIndex.map { case (value, index) =>
+        if (value == "" || value == null) s"C$index" else value
+      }
     } else {
       firstRow.zipWithIndex.map { case (value, index) => s"C$index" }
```

Comment:
I talked to @marmbrus offline. [Elsewhere](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala#L523) we use `_c0` instead of `C0`, so we should make that consistent. You're gonna have to change the name both here and in L65.